### PR TITLE
Ruby 2.6

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Ruby has been updated to 2.6.3 in macOS Catalina so let's use it everywhere.

CC @sjackman that we'll need the ARM bottles; I don't have any way of building
them.

Migrated from https://github.com/Homebrew/brew/pull/6554 as I need to debug GitHub Actions CI issues.

Fixes https://github.com/Homebrew/brew/issues/6551

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/cmd/vendor-install.sh [hunk 1]
```diff
@@ -17,17 +17,17 @@ if [[ -n "$HOMEBREW_MACOS" ]]
...
+    ruby_SHA="ab81211a2052ccaa6d050741c433b728d0641523d8742eef23a5b450811e5104"
   fi
 elif [[ -n "$HOMEBREW_LINUX" ]]
 then
   case "$HOMEBREW_PROCESSOR" in
     x86_64)
-      ruby_URL="$HOMEBREW_BOTTLE_DOMAIN/bottles-portable-ruby/portable-ruby-2.3.7.x86_64_linux.bottle.tar.gz"
-      ruby_URL2="https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.3.7/portable-ruby-2.3.7.x86_64_linux.bottle.tar.gz"
-      ruby_SHA="9df214085a0e566a580eea3dd9eab14a2a94930ff74fbf97fb1284e905c8921d"
+      ruby_URL="$HOMEBREW_BOTTLE_DOMAIN/bottles-portable-ruby/portable-ruby--2.6.3.x86_64_linux.bottle.tar.gz"
```
Comment (MindTooth) [inline Library/Homebrew/cmd/vendor-install.sh]: Is the double dash correct?  After `portable-ruby`.

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/vendor-install.sh]: Yes

**Comment by sjackman** [comment]
> @sjackman that we'll need the ARM bottles; I don't have any way of building them.

Feel free to mere without the ARM bottles, and I'll take a shot at building them tonight.

**Comment by MikeMcQuaid** [comment]
@sjackman Check the CI, unfortunately even the x86_64 bottles from Docker aren't working on Linux.

**Comment by sjackman** [comment]
```
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661: [BUG] Segmentation fault at 0x0000000000000018
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]

-- Control frame information -----------------------------------------------
c:0012 p:0008 s:0056 e:000053 BLOCK  /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661 [FINISH]
c:0011 p:---- s:0049 e:000048 IFUNC 
c:0010 p:---- s:0046 e:000045 CFUNC  :each
c:0009 p:---- s:0043 e:000042 CFUNC  :each_with_index
c:0008 p:0006 s:0039 e:000038 METHOD /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660
c:0007 p:0024 s:0034 e:000033 METHOD /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1555
c:0006 p:0047 s:0028 e:000027 METHOD /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1421
c:0005 p:0006 s:0023 e:000022 BLOCK  /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb: [FINISH]
c:0004 p:---- s:0020 e:000019 CFUNC  :synchronize
c:0003 p:0223 s:0016 e:000015 METHOD /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:
c:0002 p:0027 s:0006 e:000005 TOP    <internal:gem_prelude>:4 [FINISH]
c:0001 p:0000 s:0003 E:001620 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
<internal:gem_prelude>:4:in `<internal:gem_prelude>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1421:in `activate'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1555:in `add_self_to_load_path'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `load_path_insert_index'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `each_with_index'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661:in `block in load_path_insert_index'

-- Machine register context ------------------------------------------------
 RIP: 0x00000000006b2d0e RBP: 0x000000000000100f RSP: 0x00007ffe8826bcc0
 RAX: 0x0000000000000000 RBX: 0x0000000000000000 RCX: 0x000000000000004b
 RDX: 0x0000000002c6e428 RDI: 0x0000000000000000 RSI: 0x000000000000100f
  R8: 0x00007f5aa2f9c141  R9: 0x0000000002d33b70 R10: 0x00007f5aa309bde0
 R11: 0x0000000000000000 R12: 0x0000000000000000 R13: 0x000000000000000f
 R14: 0x00007ffe8826bd08 R15: 0x00007f5aa309bd70 EFL: 0x0000000000010246

-- C level backtrace information -------------------------------------------
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6d08b3) [0x6d08b3]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6d0abb) [0x6d0abb]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x58aef1) [0x58aef1]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6621ce) [0x6621ce]
/lib/x86_64-linux-gnu/libpthread.so.0(__restore_rt+0x0) [0x7f5aa3fc8390]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6b2d0e) [0x6b2d0e]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_callable_method_entry+0x12) [0x6b7312]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_search_method_slowpath+0x1f) [0x6b734f]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6c39aa) [0x6c39aa]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_exec+0x18a) [0x6bb95a]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_yield_values2+0x278) [0x6be828]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x5803b2) [0x5803b2]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6ba28b) [0x6ba28b]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6be1f3) [0x6be1f3]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_yield+0x30) [0x6c0690]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_ary_each+0x3d) [0x54190d]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_call0+0x36e) [0x6c100e]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6b8222) [0x6b8222]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_block_call+0x29) [0x6b8469]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x57b092) [0x57b092]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6b5742) [0x6b5742]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6bf4d6) [0x6bf4d6]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6c05db) [0x6c05db]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6c5a35) [0x6c5a35]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_exec+0x18a) [0x6bb95a]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_yield+0x258) [0x6c08b8]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_ensure+0xd2) [0x5914c2]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6b5742) [0x6b5742]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6bf4d6) [0x6bf4d6]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6c05db) [0x6c05db]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6c5a35) [0x6c5a35]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_exec+0x18a) [0x6bb95a]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x6d9483) [0x6d9483]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x661181) [0x661181]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(ruby_process_options+0x83) [0x661aa3]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(ruby_options+0xc3) [0x592703]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x478351) [0x478351]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7f5aa30bd830]
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x478389) [0x478389]

-- Other runtime information -----------------------------------------------

* Loaded script: /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby

* Loaded features:

    0 enumerator.so
    1 thread.rb
    2 rational.so
    3 complex.so
    4 encdb.so
    5 trans/transdb.so
    6 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/x86_64-linux/rbconfig.rb
    7 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/compatibility.rb
    8 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/defaults.rb
    9 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/deprecate.rb
   10 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/errors.rb
   11 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/version.rb
   12 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/requirement.rb
   13 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/platform.rb
   14 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/basic_specification.rb
   15 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/stub_specification.rb
   16 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/delegate.rb
   17 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/rfc2396_parser.rb
   18 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/rfc3986_parser.rb
   19 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/common.rb
   20 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/generic.rb
   21 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/file.rb
   22 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ftp.rb
   23 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/http.rb
   24 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/https.rb
   25 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ldap.rb
   26 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ldaps.rb
   27 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/mailto.rb
   28 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri.rb
   29 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification_policy.rb
   30 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/util/list.rb
   31 stringio.so
   32 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb
   33 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/exceptions.rb
   34 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/util.rb
   35 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/bundler_version_finder.rb
   36 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/dependency.rb
   37 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb
   38 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/monitor.rb
   39 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb
   40 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_warn.rb
   41 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb
   42 /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/path_support.rb

* Process memory map:

00400000-00d58000 r-xp 00000000 08:01 3638817                            /
00f57000-00f7b000 rwxp 00957000 08:01 3638817                            /
00f7b000-00f97000 rwxp 00000000 00:00 0 
02bf7000-03014000 rwxp 00000000 00:00 0                                  [heap]
7f5aa0145000-7f5aa030e000 r-xs 00000000 08:01 2863203                    /
7f5aa030e000-7f5aa0d7d000 r-xs 00000000 08:01 3638817                    /
7f5aa0d7d000-7f5aa0d93000 r-xp 00000000 08:01 2863224                    /
7f5aa0d93000-7f5aa0f92000 ---p 00016000 08:01 2863224                    /
7f5aa0f92000-7f5aa0f93000 rwxp 00015000 08:01 2863224                    /
7f5aa0f93000-7f5aa309d000 rwxp 00000000 00:00 0 
7f5aa309d000-7f5aa325d000 r-xp 00000000 08:01 2863203                    /
7f5aa325d000-7f5aa345d000 ---p 001c0000 08:01 2863203                    /
7f5aa345d000-7f5aa3461000 r-xp 001c0000 08:01 2863203                    /
7f5aa3461000-7f5aa3463000 rwxp 001c4000 08:01 2863203                    /
7f5aa3463000-7f5aa3467000 rwxp 00000000 00:00 0 
7f5aa3467000-7f5aa3469000 r-xp 00000000 08:01 2863297                    /
7f5aa3469000-7f5aa3668000 ---p 00002000 08:01 2863297                    /
7f5aa3668000-7f5aa3669000 r-xp 00001000 08:01 2863297                    /
7f5aa3669000-7f5aa366a000 rwxp 00002000 08:01 2863297                    /
7f5aa366a000-7f5aa3772000 r-xp 00000000 08:01 2863235                    /
7f5aa3772000-7f5aa3971000 ---p 00108000 08:01 2863235                    /
7f5aa3971000-7f5aa3972000 r-xp 00107000 08:01 2863235                    /
7f5aa3972000-7f5aa3973000 rwxp 00108000 08:01 2863235                    /
7f5aa3973000-7f5aa397c000 r-xp 00000000 08:01 2863211                    /
7f5aa397c000-7f5aa3b7b000 ---p 00009000 08:01 2863211                    /
7f5aa3b7b000-7f5aa3b7c000 r-xp 00008000 08:01 2863211                    /
7f5aa3b7c000-7f5aa3b7d000 rwxp 00009000 08:01 2863211                    /
7f5aa3b7d000-7f5aa3bab000 rwxp 00000000 00:00 0 
7f5aa3bab000-7f5aa3bae000 r-xp 00000000 08:01 2863216                    /
7f5aa3bae000-7f5aa3dad000 ---p 00003000 08:01 2863216                    /
7f5aa3dad000-7f5aa3dae000 r-xp 00002000 08:01 2863216                    /
7f5aa3dae000-7f5aa3daf000 rwxp 00003000 08:01 2863216                    /
7f5aa3daf000-7f5aa3db6000 r-xp 00000000 08:01 2863277                    /
7f5aa3db6000-7f5aa3fb5000 ---p 00007000 08:01 2863277                    /
7f5aa3fb5000-7f5aa3fb6000 r-xp 00006000 08:01 2863277                    /
7f5aa3fb6000-7f5aa3fb7000 rwxp 00007000 08:01 2863277                    /
7f5aa3fb7000-7f5aa3fcf000 r-xp 00000000 08:01 2863271                    /
7f5aa3fcf000-7f5aa41ce000 ---p 00018000 08:01 2863271                    /
7f5aa41ce000-7f5aa41cf000 r-xp 00017000 08:01 2863271                    /
7f5aa41cf000-7f5aa41d0000 rwxp 00018000 08:01 2863271                    /
7f5aa41d0000-7f5aa41d4000 rwxp 00000000 00:00 0 
7f5aa41d4000-7f5aa41fa000 r-xp 00000000 08:01 2863183                    /
7f5aa422e000-7f5aa4250000 r-xs 00000000 08:01 2863271                    /
7f5aa4250000-7f5aa4257000 r-xs 00000000 08:01 2864074                    /
7f5aa4257000-7f5aa43ef000 r-xp 00000000 08:01 3380710                    /
7f5aa43ef000-7f5aa43f4000 rwxp 00000000 00:00 0 
7f5aa43f8000-7f5aa43f9000 rwxp 00000000 00:00 0 
7f5aa43f9000-7f5aa43fa000 r-xp 00025000 08:01 2863183                    /
7f5aa43fa000-7f5aa43fb000 rwxp 00026000 08:01 2863183                    /
7f5aa43fb000-7f5aa43fc000 rwxp 00000000 00:00 0 
7ffe87a77000-7ffe88276000 rwxp 00000000 00:00 0                          [stack]
7ffe883c3000-7ffe883c6000 r--p 00000000 00:00 0                          [vvar]
7ffe883c6000-7ffe883c7000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]


[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: https://www.ruby-lang.org/bugreport.html

Error: Forked child process failed: pid 19720 SIGABRT (signal 6) (core dumped)
```
https://github.com/Homebrew/brew/pull/6556/checks?check_run_id=252191345#step:5:438

**Comment by sjackman** [comment]
How exciting. It's failing with a `SIGABRT`, which is usually intentional by the program itself calling `abort()`. The `C level backtrace information` doesn't appear complete though. I'll take a look into it tonight.

**Comment by MikeMcQuaid** [comment]
@sjackman Perfect, thanks. Given no-one is using these bottles yet I feel safe unpublishing/republishing them if necessary (if we need, as I suspect, `homebrew-portable-ruby` changes).

**Comment by reitermarkus** [comment]
> most files are not changed

There's no problem with that. What are you trying to say here?

**Comment by sjackman** [comment]
```
Failed examples:

rspec ./test/cmd/reinstall_spec.rb:11 # brew reinstall reinstalls a Formula
rspec ./test/cmd/migrate_spec.rb:10 # brew migrate migrates a renamed Formula
rspec ./test/cmd/upgrade_spec.rb:10 # brew upgrade upgrades a Formula and cleans up old versions
rspec ./test/cmd/install_spec.rb:10 # brew install installs formulae
rspec ./test/cmd/install_spec.rb:19 # brew install can install keg-only Formulae
rspec ./test/cmd/install_spec.rb:32 # brew install can install HEAD Formulae
rspec ./test/dev-cmd/ruby_spec.rb:10 # brew ruby executes ruby code with Homebrew's libraries loaded
```

**Comment by sjackman** [comment]
I can reproduce `brew tests` failing for these seven tests in Docker on my Macbook Pro.
`brew install hello` works as expected, as does `brew install -s hello`.
That's as far as I've gotten so far. I'm not yet sure why `brew install` is failing within `brew tests`.

**Comment by MikeMcQuaid** [comment]
> I can reproduce `brew tests` failing for these seven tests in Docker on my Macbook Pro.
> `brew install hello` works as expected, as does `brew install -s hello`.
> That's as far as I've gotten so far. I'm not yet sure why `brew install` is failing within `brew tests`.

@sjackman I think we need to rebuild the bottle in a different configuration if it's failing on the C level like this at all. I'm also open to options for how best to handle this given it's currently blocking us matching the Ruby version on macOS which means different behaviour on different platforms (and `brew tests` failing there).

**Comment by MikeMcQuaid** [comment]
CC @mistydemeo for her wizardry and general smarts

**Comment by sjackman** [comment]
I'm cautiously optimistic that PR https://github.com/Homebrew/brew/pull/6569 resolves the broken `portable-ruby` 2.6.3 bottle on Linux. The bottle will need to be rebuilt after merging that PR.

**Comment by MikeMcQuaid** [comment]
> I'm cautiously optimistic that PR #6569 resolves the broken `portable-ruby` 2.6.3 bottle on Linux. The bottle will need to be rebuilt after merging that PR.

Unfortunately not. It's not a `patchelf` issue it seems.

**Pull request merged**